components/dropdown/dropdown.test.js

```diff
@@ -81,148 +62,132 @@ describe('DtDropdown Tests', () => {
     testContext.localVue = createLocalVue();
   });
 
-  // Teardown
-  afterEach(() => {
-    propsData = basePropsData;
-    slots = baseSlots;
-    scopedSlots = baseScopedSlots;
-    listeners = {};
-    wrapper.destroy();
-  });
-
   afterAll(() => {
     // Restore RequestAnimationFrame and cancelAnimationFrame
     global.requestAnimationFrame = undefined;
     global.cancelAnimationFrame = undefined;
   });
 
+  beforeEach(() => {
+    updateWrapper();
+  });
+
+  afterEach(() => {
+    mockProps = {};
+    mockSlots = {};
+    mockScopedSlots = {};
+    mockListeners = {};
+  });
+
   describe('Presentation Tests', () => {
-    // Test setup
-    beforeEach(() => {
-      _setWrappers();
+    it('should render the component', () => {
+      expect(wrapper.exists()).toBe(true);
     });
 
-    it(
-      'should render the component',
-      () => { expect(wrapper.exists()).toBe(true); },
-    );
+    it('should not render the visually hidden close button', () => {
+      const buttonExists = wrapper.findComponent(SrOnlyCloseButton).exists();
 
-    it('should not render the visually hidden close button', async () => {
-      itBehavesLikeVisuallyHiddenCloseButtonExists(wrapper, false);
+      expect(!buttonExists).toBe(true);
     });
 
     describe('When a list is provided', () => {
-      it(
-        'should render the list wrapper',
-        () => { expect(listWrapper.exists()).toBe(true); },
-      );
-      it(
-        'should render the anchor',
-        () => { expect(anchorElement.exists()).toBe(true); },
-      );
-      it(
-        'should render the list',
-        () => { expect(wrapper.find('#list').exists()).toBe(true); },
-      );
+      it('should render the list wrapper', () => {
+        expect(listWrapper.exists()).toBe(true);
+      });
+
+      it('should render the anchor', () => {
+        expect(anchorElement.exists()).toBe(true);
+      });
+
+      it('should render the list', () => {
+        expect(wrapper.find('#list').exists()).toBe(true);
+      });
     });
 
     describe('When visuallyHiddenClose is true', () => {
-      beforeEach(async () => {
-        await wrapper.setProps({ visuallyHiddenClose: true });
+      beforeEach(() => {
+        mockProps = { visuallyHiddenClose: true };
+
+        updateWrapper();
       });
 
       it('should contain a visually hidden close button', () => {
-        itBehavesLikeVisuallyHiddenCloseButtonExists(wrapper);
+        const buttonExists = wrapper.findComponent(SrOnlyCloseButton).exists();
+
+        expect(buttonExists).toBe(true);
```

Comment:
Reuse this by adding it to `updateWrapper`

Reply:
Updated!